### DISCUSSION TITLE
HTML and CSS Help

### DISCUSSION THREAD

**Discussion by Ethan-Mircheff** [community/community#141116] [Programming Help] [answered]
### Body

I am trying to make a GitHub pages site for a photography portfolio, but I am having an issue with the spacing of the images in one of the subpages.  Is there a good way to fix this? It only happens on computers, on Edge mobile and Safari on iPhone it works perfectly.
Repository: [https://github.com/Ethan-Mircheff/photography](https://github.com/Ethan-Mircheff/photography)
Site: [https://ethan-mircheff.github.io/photography/](https://ethan-mircheff.github.io/photography/)

![Screenshot 2024-10-10 5 48 31 PM](https://github.com/user-attachments/assets/4639523f-30f6-4d13-8595-aa6187fb9435)

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by yunwei-zh** [comment]
Hi Ethan,

Thanks for your detailed description.

After reading your post carefully, I noticed what you are facing now.

When you reduce the window size, the images are only in three columns, and one column.

Not two columns.

You will have to fix this to complete it perfectly.

That could be an issue.

To fix it, you need to add more variables to control precisely.

If you are interested, please DM

Hope to contact with you soon.

Best regards

**Reply by Ethan-Mircheff** [reply]
Thanks for your reply, I would like some help with the code to solve this, I did some research but I wasn't able to find any good sources before.  What variables do you recommend adding to fix the scaling?

**Comment by ricardious** [comment]
Try using `display: flex` or `grid` on the image container with `gap` for consistent spacing, and ensure `max-width` is set for images. Adjust margins using media queries for larger screens.

**Reply by Ethan-Mircheff** [reply]
Could you give me an example of where to implement this? I am still fairly new to CSS.

**Reply by ricardious** [reply, marked as answer]
Sorry, my English is not so good. For fixing the spacing issue, you can use **flexbox** instead of `float`. This helps with better spacing between images. Here’s an example:

```css
.row {
  display: flex;
  flex-wrap: wrap;
  gap: 10px;
}

.pic {
  flex: 1 1 30%;
  max-width: 30%;
  padding: 0;
}

@media screen and (max-width: 600px) {
  .pic {
    flex: 1 1 100%;
    max-width: 100%;
  }
}
```

I hope you mark my answer as correct.

**Reply by Ethan-Mircheff** [reply]
Thanks for the help, that worked great.
